pkg/ingester/client/ingester.proto

```diff
@@ -76,50 +76,7 @@ message LabelValueSeriesCount {
   map<string, uint64> label_value_series = 2;
 }
 
-message ReadRequest {
-  repeated QueryRequest queries = 1;
-
-  enum ResponseType {
-    SAMPLES = 0;
-    STREAMED_XOR_CHUNKS = 1;
-  }
-  repeated ResponseType accepted_response_types = 2;
-}
-
-message ReadResponse {
-  repeated QueryResponse results = 1;
-}
-
-message StreamReadResponse {
-  repeated StreamChunkedSeries chunked_series = 1;
-
-  int64 query_index = 2;
-}
-
-message StreamChunkedSeries {
-  repeated cortexpb.LabelPair labels = 1 [(gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/grafana/mimir/pkg/mimirpb.LabelAdapter"];
-  repeated StreamChunk chunks = 2 [(gogoproto.nullable) = false];
-}
-
-message StreamChunk {
-  int64 min_time_ms = 1;
-  int64 max_time_ms = 2;
-
-  enum Encoding {
-    UNKNOWN = 0;
-    XOR     = 1;
-    HISTOGRAM = 2;
-    FLOAT_HISTOGRAM = 3;
-  }
-  Encoding type  = 3;
-  bytes data     = 4 [(gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/grafana/mimir/pkg/mimirpb.UnsafeByteSlice"];
-}
-
 message QueryRequest {
-  // This QueryRequest message is also used for remote read requests, which includes a hints field we don't support.
```

Comment:
Note to reviewers: previously, `QueryRequest` was referenced by `ReadRequest` so it was right that the field 4 (hints) was reserved. After this PR, `QueryRequest` is just used internally in Mimir and it's used anymore to decode a remote read request, so we don't have any Prometheus-compatibility requirement anymore.